commit c337104b1a16becc486fdc95d544c835b17021db
Author: Pierre-Louis Bossart <pierre-louis.bossart@linux.dev>
Date:   Sat Dec 15 14:07:22 2018 -0600

    ALSA: HD-Audio: SKL+: abort probe if DSP is present and Skylake driver selected
    
    Now that the SST/Skylake driver supports per platform selectors, we
    can add logic to automatically select the right driver.
    
    If the Skylake driver is selected for a specific platform, and the DSP
    is detected at run-time based on the PCI class/subclass/prog-if
    information, the legacy HDaudio driver aborts the probe. This will
    result in a single driver probing and remove the need for modprobe
    blacklists.
    
    Follow-up patches will add a module parameter to bypass the logic if
    this automatic detection fails, or if the Skylake driver is unable to
    actually support the platform (firmware authentication, missing
    topology file, hardware issue, etc).
    
    The same mechanism will be used to conflicts generated by the same PCI
    ID being registered by both legacy HDAuudio and SOF drivers for Intel
    platforms. In other words SOF will not require changes to the HDaudio
    legacy.
    
    Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
    Signed-off-by: Takashi Iwai <tiwai@suse.de>

diff --git a/sound/soc/intel/Kconfig b/sound/soc/intel/Kconfig
index 99a62ba409df..2fd1b61e8331 100644
--- a/sound/soc/intel/Kconfig
+++ b/sound/soc/intel/Kconfig
@@ -188,6 +188,12 @@ config SND_SOC_INTEL_SKYLAKE_COMMON
 	select SND_SOC_TOPOLOGY
 	select SND_SOC_INTEL_SST
 	select SND_SOC_HDAC_HDA if SND_SOC_INTEL_SKYLAKE_HDAUDIO_CODEC
+	select SND_HDA_INTEL_DSP_DETECTION_SKL if SND_SOC_INTEL_SKL
+	select SND_HDA_INTEL_DSP_DETECTION_APL if SND_SOC_INTEL_APL
+	select SND_HDA_INTEL_DSP_DETECTION_KBL if SND_SOC_INTEL_KBL
+	select SND_HDA_INTEL_DSP_DETECTION_GLK if SND_SOC_INTEL_GLK
+	select SND_HDA_INTEL_DSP_DETECTION_CNL if SND_SOC_INTEL_CNL
+	select SND_HDA_INTEL_DSP_DETECTION_CFL if SND_SOC_INTEL_CFL
 	select SND_SOC_ACPI_INTEL_MATCH
 	help
 	  If you have a Intel Skylake/Broxton/ApolloLake/KabyLake/